.github/workflows/demo-more-modules.yaml

```diff
@@ -0,0 +1,34 @@
+name: Build more-modules app
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+# Minimum permissions required by skip-duplicate-actions
+permissions:
+  actions: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: [ ubuntu-latest ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: semeru
+
+      - id: install-mvnd
+        uses: ./.github/actions/install-mvnd
+
+      - name: Build application with Maven
+        shell: bash
+        run: |
+          cd demo-resources/app-more-modules || exit 1
```

Comment:
How about [setting the working directory through config](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/setting-a-default-shell-and-working-directory#setting-default-shell-and-working-directory)?